packages/plugin-ext/src/plugin/documents.ts

```diff
@@ -238,13 +238,14 @@ export class DocumentsExtImpl implements DocumentsExt {
         }
     }
 
-    async openDocument(uri: URI): Promise<DocumentDataExt | undefined> {
+    async openDocument(uri: URI, options?: { language?: string; content?: string; encoding?: string }): Promise<DocumentDataExt | undefined> {
+        // If we have the document cached and no encoding options are provided,
+        // we should just return current document
         const cached = this.editorsAndDocuments.getDocument(uri.toString());
-        if (cached) {
+        if (cached && !options?.encoding) {
```

Comment:
Why does passing an encoding invalidate the cache? Maybe we read the content with the same encoding before?